python/ray/tune/tests/test_syncer_callback.py

```diff
@@ -211,6 +213,29 @@ def test_syncer_callback_sync(ray_start_2_cpus, temp_data_dirs):
     assert_file(True, tmp_target, "subdir_exclude/something/somewhere.txt")
 
 
+def test_syncer_callback_sync_with_invalid_ip(ray_start_2_cpus, temp_data_dirs):
+    """Check that the sync client updates the IP correctly"""
+    tmp_source, tmp_target = temp_data_dirs
+
+    syncer_callback = TestSyncerCallback(local_logdir_override=tmp_target)
+
+    trial1 = MockTrial(trial_id="a", logdir=tmp_source)
+
+    syncer_callback._trial_ips[trial1.trial_id] = "invalid"
+    syncer_callback.on_trial_start(iteration=0, trials=[], trial=trial1)
+
+    syncer_callback.on_trial_result(iteration=1, trials=[], trial=trial1, result={})
```

Comment:
I am a bit confused as to why would we encounter this?
if a result is reported, shouldn't there always exist a NODE_IP field, which is automatically filled in by us?

Reply:
The problem is that the `last_result` could be populated from a restored checkpoint and point to an IP address that does not exist anymore